crates/oro-manifest/src/lib.rs

```diff
@@ -0,0 +1,324 @@
+    pub name: Option<String>,
+
+    #[builder(setter(strip_option), default)]
+    pub version: Option<Version>,
+
+    #[builder(setter(into, strip_option), default)]
+    pub description: Option<String>,
+
+    #[builder(setter(into, strip_option), default)]
+    pub homepage: Option<String>,
+
+    #[serde(default, alias = "licence")]
+    #[builder(setter(into, strip_option), default)]
+    pub license: Option<String>,
+
+    #[serde(default)]
+    #[builder(setter(strip_option), default)]
+    pub browser: bool,
+
+    #[serde(default)]
+    #[builder(setter(strip_option), default)]
+    pub bugs: Option<Bugs>,
+
+    #[serde(default)]
+    #[builder(default)]
+    pub keywords: Vec<String>,
+
+    #[builder(setter(strip_option), default)]
+    pub bin: Option<Bin>,
+
+    #[serde(default)]
+    #[builder(setter(strip_option), default)]
+    pub author: Option<PersonField>,
+
+    #[serde(default)]
+    #[builder(default)]
+    pub contributors: Vec<PersonField>,
+
+    #[serde(default)]
+    #[builder(default)]
+    pub files: Vec<String>,
+
+    #[builder(setter(into, strip_option), default)]
+    pub main: Option<String>,
+
+    #[builder(setter(strip_option), default)]
+    pub man: Option<Man>,
+
+    #[serde(default)]
+    #[builder(default)]
+    pub directories: Directories,
+
+    #[serde(rename = "type")]
+    #[builder(setter(into, strip_option), default)]
+    pub module_type: Option<String>,
+
+    #[builder(setter(strip_option), default)]
+    pub exports: Option<Exports>,
+
+    #[builder(setter(strip_option), default)]
+    pub imports: Option<Imports>,
+
+    #[builder(setter(strip_option), default)]
+    pub repository: Option<Repository>,
+
+    #[serde(default)]
+    #[builder(default)]
+    pub scripts: HashMap<String, String>,
+
+    #[builder(setter(strip_option), default)]
+    pub config: Option<serde_json::Value>,
+
+    #[serde(default)]
+    #[builder(default)]
+    pub os: Vec<String>,
+
+    #[serde(default)]
+    #[builder(default)]
+    pub cpu: Vec<String>,
+
+    #[serde(default)]
+    #[builder(default)]
+    pub private: bool,
+
+    #[serde(default)]
+    #[builder(default)]
+    pub publish_config: HashMap<String, String>,
+
+    // Deps
+    #[serde(default)]
+    #[builder(default)]
+    pub dependencies: HashMap<String, VersionReq>,
+
+    #[serde(default)]
+    #[builder(default)]
+    pub dev_dependencies: HashMap<String, VersionReq>,
+
+    #[serde(default)]
+    #[builder(default)]
+    pub optional_dependencies: HashMap<String, VersionReq>,
+
+    #[serde(default)]
+    #[builder(default)]
+    pub peer_dependencies: HashMap<String, VersionReq>,
+
+    #[serde(default, alias = "bundleDependencies", alias = "bundledDependencies")]
+    #[builder(default)]
+    pub bundled_dependencies: Vec<String>,
+
+    #[serde(default)]
+    #[builder(default)]
+    pub workspaces: Vec<String>,
+
+    #[serde(flatten, default)]
+    #[builder(default)]
+    pub _rest: HashMap<String, serde_json::Value>,
```

Comment:
We might consider using a `BTreeMap` here? (`HashMap` will randomly order the rest of the keys, while `BTreeMap` will order them lexically, which is stable if we're roundtripping edits through this tool.)

Reply:
See discussion in Discord. tl;dr: I don't think we should use `OroManifest` as the roundtripping datastructure, because it _already_ establishes an order. Instead, we should use `serde_json::Value` with the appropriate feature to retain value ordering when we write stuff back out.